editor/editor_properties_array_dict.cpp

```diff
@@ -125,13 +125,13 @@ EditorPropertyDictionaryObject::EditorPropertyDictionaryObject() {
 
 ///////////////////// ARRAY ///////////////////////////
 
-void EditorPropertyArray::_property_changed(const String &p_prop, Variant p_value) {
+void EditorPropertyArray::_property_changed(const String &p_prop, Variant p_value, bool changing) {
 
 	if (p_prop.begins_with("indices")) {
 		int idx = p_prop.get_slice("/", 1).to_int();
 		Variant array = object->get_array();
 		array.set(idx, p_value);
-		emit_signal("property_changed", get_edited_property(), array);
+		emit_signal("property_changed", get_edited_property(), array, true);
```

Comment:
I tried setting this to `changing` and the cursor stucks while editing Vector2 by dragging.

Reply:
Yeah, you don't need to create `bool changing` for this method. Just pass `true` to the signal like I did here: https://github.com/godotengine/godot/pull/19351/files#diff-646237aede31b718d50a4d2b41b0e7fcR53

Reply:
In my PR, I didn't change the signature of the method `_text_changed(const String &p_string)` to `_text_changed(const String &p_string, bool changing)` because this is not needed. 

I think it's better to change 
`EditorPropertyArray::_property_changed(const String &p_prop, Variant p_value, bool changing) {` 
to
`EditorPropertyArray::_property_changed(const String &p_prop, Variant p_value) {` because this will not be used.

Reply:
Thanks! I'll try if I can remove the `bool changing` in the function definition.

Reply:
I tried removing `bool changing` in both .cpp and .h and got this error:
```
ERROR: emit_signal: Error calling method from signal 'property_changed': 'EditorPropertyDictionary::_property_changed': Method expected 2 arguments, but called with 3.
   At: core/object.cpp:1261.
```
I got a compile error when removing from cpp only.

Reply:
If you want to remove it, it has to be removed everywhere. Which asks the question of why it was added in the first place? I guess there would be some logic which depends on it in some cases.

Reply:
@Noshyaar You're right, we can't remove this param. I was comparing this file with `editor_property.cpp` but editor_property don't have any `property_changed`, this is why I didn't needed to made that kind of change.

> I tried setting this to changing and the cursor stucks while editing Vector2 by dragging.

I tested here on linux and I didn't have this problem. What `Vector2` you tested?

Reply:
Currently, emiting with `true` as the final parameter, works for me. The cursor doesn't stuck anymore.

Reply:
@akien-mga The `changing` is used to prevent the inspector tree to be updated while you are typing for example, otherwise we get a reversed typed.
https://github.com/godotengine/godot/blob/16f7ff1ee946cc04f03670fb66eeb24392d8384d/editor/editor_inspector.cpp#L1874-L1887

Reply:
@Noshyaar I think what you did is right. I think it best to leave it as it is.

Reply:
@guilhermefelipecgs I'm not sure it's the best way to fix this issue though, it was hard to review amidst the other changes in 709948aa5, but I think @reduz might have had objections to adding this parameter to `_property_changed` (especially given follow-up issues such as those linked in OP). I'm not saying it should be undone, but it might be worth looking into a better solution.